SwissKnife/gcloud/GCloudStorage.py

```diff
@@ -0,0 +1,159 @@
+                    origin_data_str: str,
+                    destination_file_name: str,
+                    destination_storage_path: str,
+                    encoding: str=None) -> str:
+        """Transforms a string to a text file and uploads it to
+        gcloud Storage.
+
+        :param origin_data_str: String to save in a file
+        :type origin_data_str: str
+        :param destination_file_name: Name that it will have in Storage
+        :type destination_file_name: str
+        :param destination_storage_path: Storage path were it'll be stored
+        :type destination_storage_path: str
+        :param encoding: Specific encoding of the file, defaults to None
+        :type encoding: str, optional
+        :return: Storage path were the file has been uploaded
+        :rtype: str
+        """
+        return self.__save_to_storage(data=origin_data_str,
+                                      data_encoding=encoding,
+                                      data_type='str',
+                                      destionation_storage_path=destination_file_name,
+                                      destionation_file_name=destination_storage_path
+                                      )
+    
+    def __save_to_storage(self,
+                          data,
+                          data_encoding: str,
+                          data_type: str,
+                          destionation_storage_path: str,
+                          destionation_file_name: str) -> str:
+        """Uploads data, which may come in several formats, to
+        Storage. Accepted data types are 'file' and 'str', matching
+        methods save_file and save_string of this same class.
+
+        :param data: Data file or string that will be uploaded
+        :type data: [type]
+        :param data_encoding: Specific encoding of the file
+        :type data_encoding: str
+        :param data_type: Type of data to upload: file or string
+        :type data_type: str
+        :param destionation_storage_path: Storage path were it'll be stored
+        :type destionation_storage_path: str
+        :param destionation_file_name: Name that it will have in Storage
+        :type destionation_file_name: str
+        :raises NotImplementedError: When the data type provided is not implemented
+        :return: gcloud storage of the uploaded file
+        :rtype: str
+        """
+        blob = self.__generate_blob(destionation_storage_path, destionation_file_name, data_encoding)
+        
+        if type == 'str':
+            blob.upload_from_string(data)
+        elif type == 'file':
+            blob.upload_from_filename(data)
+        else:
+            raise NotImplementedError
+        
+        file_path = self._get_storage_file_path(destionation_storage_path, destionation_file_name, with_bucket=True)
+        
+        self.logger.info(f"Uploaded file to gcloud path {file_path}")
+        return file_path
+        
+    def __generate_blob(self, 
```

Comment:
The return type is missing. If it is complicated to defined the return type, you can set a descriptive string:

`def _generate_blob(...) -> "blob":`

Reply:
Nice, I'll do that